engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -512,7 +591,28 @@ class TableScan : public kernel {
 	}
 
 private:
-	DataSourceSequence input; /**< Input data source sequence. */
+	std::shared_ptr<ral::io::data_provider> provider;
+	std::shared_ptr<ral::io::data_parser> parser;
+	ral::io::Schema  schema; /**< Table schema associated to the data to be loaded. */ 
+	size_t file_index = 0;
```

Comment:
Ok, so the way to fix this is to set current_batch = this->output_.total_batches_added();
`total_batches_added()` will be a new function for the port, similar to `total_rows_added()` and it will get the num batches from the output CacheMachine using `get_num_batches`